### ISSUE TITLE
"No dimensions set for key window" error after fresh 0.44.0 install

### ISSUE THREAD

**Issue by EnriqueDev** [facebook/react-native#13758]
### Description

After a fresh install of react native the following error appears on screen:

![captura de pantalla 2017-05-03 a las 12 43 26](https://cloud.githubusercontent.com/assets/25365731/25659120/2b483e00-2ffe-11e7-87a2-b49dd9750221.png)

### Reproduction Steps and Sample Code
 
1. Install fresh project using: react-native init newProject
2. Run on ios simulator: react-native run-ios

### Additional Information

* React Native version: 0.44.0
* Platform: iOS
* Development Operating System: MacOS Sierra
* Dev tools: default

**Comment by jbreuer95** [comment]
+1

**Comment by andrii-yarmola** [comment]
same issue on run-android
win7

**Comment by NickJorgensen** [comment]
same here
react native 0.44.0
macOS 10.12 Sierra
xcode 8.3.2

The error does not show up when building "Release" on a device, only "Debug".

**Comment by choiway** [comment]
+1

**Comment by Amurmurmur** [comment]
Sounds strange, but doesnt happen for me. 
I use Dimensions lib and it runs just fine over couple of RN releases right from 0.39 up to 0.44.

**Comment by rawrmaan** [comment]
Is it possible that any of you are using `NavigationExperimental` in your app? This module was removed in 0.44 and can cause this error if your app tries to require it.

**Comment by choiway** [comment]
@rawrmaan This happens on a fresh install.

**Comment by andrii-yarmola** [comment]
@rawrmaan Not quite
in my case fresh app is working good,
i just installed "react-native-router-flux" with npm (so probably you are right)
and i got this error w/o code changes

**Comment by jebrial** [comment]
Try closing terminal, deleting ios/build and run again. I had the same issue but now its working.

**Comment by hanih** [comment]
try importing StaticContainer from react native

**Comment by hellomaya** [comment]
For the case of android, try close terminal, run ``npm cache clean`` and then by using new react-native-cli, which I removed react-native-cli from 0.42, and installed react-native-cli for 0.44,  then  run ```react-native run-android```, worked for me.

**Comment by gotoAndBliss** [comment]
+1 Same issues. Fresh install.

**Comment by mattdufeu** [comment]
I had the same error on a brand new project. Turned out I had an old Package Manager running from a different project. Once I closed that, the fresh app worked.

**Comment by GetSource1234** [comment]
Same issue. Got it on Android

**Comment by itamarborges** [comment]
Just close terminal, open a new one, run npm cache clean, and run your project again. Boom, done!

**Comment by mrbrentkelly** [comment]
On Android it seems to be an issue with using React Native **v0.44.0** along with Android SDK Build Tools version **25**.

If you downgrade to React Native 0.43.2 the issue goes away or alternatively if you downgrade your Android `compileSdkVersion` to **23** (and `buildToolsVersion` to **23.0.1**) the issue also goes away.

**Fix 1**

In your app `build.gradle` change `compile "com.facebook.react:react-native:+"` to `compile "com.facebook.react:react-native:0.43.2@aar"`

OR

**Fix 2**

In your app `build.gradle ` downgrade your SDK to target and compile against 23

```
compileSdkVersion 23
buildToolsVersion "23.0.1"
...
targetSdkVersion 23
...
compile "com.android.support:appcompat-v7:23.0.1"
```

**Comment by dcarrot2** [comment]
We're seeing this on Android and our `buildToolsVersion` is `23.0.1` and our `compileSdkVersion` is `23`.

**Comment by rkbhochalya** [comment]
Got same issue on android. I closed the old Package Manager running from a different project and it worked.

**Comment by cesarmal** [comment]
After reading all the commentaries I and trying different solutions, I got this 5 points. Maybe they are helpful for someone else.

1) The fresh app worked for me;
2) After installing react-native-router-flux, the issue came out - like @andrii-yarmola said;
3) I had others projects on the same machine - with different library versions. But the solution given by @rkbhochalya  and @mattdufeu did not work for me;
4) It happened with iOS and Android;
4) Only after **rebooting** the machine - MacOSX - the issue was gone.

**Comment by rkbhochalya** [comment]
You can also try clearing cache `watchman watch-del-all && rm -rf node_modules/ && npm cache clean && npm install && npm start -- --reset-cache` or if you have yarn installed `watchman watch-del-all && rm -rf node_modules/ && npm cache clean && yarn install && npm start -- --reset-cache`

**Comment by rkbhochalya** [comment]
Also close all package manager terminal windows running from different projects.

**Comment by Neustren** [comment]
@rkbhochalya clearing cache resolves. thanks.

**Comment by efnan34** [comment]
Same issue at fresh project.Closing terminal and clearing cache didnt work.I am totally stuck.target sdk 22 
I

**Comment by JINGTINGWU** [comment]
I get the same error on Android 6 with React Native v0.44.0，I downgrade RN version to v0.43.0， it is work.

**Comment by aleygues** [comment]
@rkbhochalya clearing cache also resolves. thanks!

**Comment by garthenweb** [comment]
`npm start -- --reset-cache` was working for me!

**Comment by Sxiaobai** [comment]
@NickJorgensen  Have you worked out any approach to solve the problem?

**Comment by Sxiaobai** [comment]
@rkbhochalya  Only when release can have this mistake  Can you solve it

**Comment by tanotify** [comment]
+1 Same error

**Comment by nonameolsson** [comment]
I had the same problem but thanks to the comments I managed to solve it.

I upgraded to:
    "react": "16.0.0-alpha.6",
    "react-native": "0.44.2",

After that:
`rm -rf $TMPDIR/react-* && watchman watch-del-all && npm cache clean`
`rm -rf node_modules`
`npm install`

That solved it.

**Comment by luco** [comment]
+1 
This is a pain.

**Comment by gadfil** [comment]
+1
on device all fine
on emulator same error

**Comment by avogel3** [comment]
+1
Deleting watchman cache and reinstalling node modules still does not fix the error for me

**Comment by liujunhongz** [comment]
node v7.10.0 was working for me, but node v8.0.0+ was not.

**Comment by Karl-Gustav** [comment]
Just running `react-native upgrade` fixed it for me

**Comment by nonameolsson** [comment]
I got this problem again, but this time 
```
rm -rf $TMPDIR/react-* && watchman watch-del-all && npm cache clean
rm -rf node_modules
npm install
```
didn't help.

After a while the project started again, but I have no idea of why!

**Comment by nigelnindo** [comment]
I run into this on iOS, when I switched react native versions. I think it happens due to stale react native binaries. Cleaning the project and running the build in Xcode solved it for me.

**Comment by roch2810** [comment]
Close all terminal 
run npm cache clean 
and after you can run your project with react-native run-ios
worked for me

**Comment by woxiangxin7779** [comment]
For the case of android, try close terminal,then run react-native run-android, worked for me.

**Comment by dereknex** [comment]
if you got issue with **Debug** build, you can fix it as above.
if you got issue with **Release** build, you need regenerate `main.jsbundle` by `react-native bundle  --minify --dev false --assets-dest ./ios --entry-file index.ios.js --platform ios --bundle-output ios/main.jsbundle`.

**Comment by samthui** [comment]
Mine is solved after downgrading npm from @5 to @4.6.1
